CrowdStrike/ember-aria
# Handling tables, async selectors

## What changed

Before, `{{aria-grid}}` only worked for `role=grid` where the rows were all loaded at once. This now fixes that so it works for when the rows are loaded later (for example, when data needs to be fetched in order to populate the table).

- [x] added table examples
- [x] added async versions of all the examples where the rows are loaded loader
- [x] updated the MutationObserver to run without anything in the WeakMap. This Map looks like it was always empty by default and so the observer was not observing at all due to an early return. Not sure how this map works, happy to put it back in if it's needed.
- [x] updated row selectors to select `tr` tags not just `role=row`.
- [x] updated `[role=cell]` selectors to also check for `td` and `th`.
- [x] update `[role=columnheader]` to also check for `th`.
- [x] created new helper components as colocated components, due to: https://github.com/embroider-build/embroider/issues/1261 
Previous failed build: https://github.com/CrowdStrike/ember-aria/actions/runs/3331223927/jobs/5510763143


## How to test
Looks like there are no test links....you might have to pull these branch and run things locally. There are some existing bugs with pnpm and turbo, so try
a) pull branch
b) pnpm i
c) pnpm build
d) `pnpm i` or, if you have to `pnpm i -f`
e) pnpm dev
f) go to http://localhost:4200/docs

1. Wait for the table to load
2. TAB should take you through the navigation and THEN to the first non-header cell.
3. ARROW KEYS should take you around the table cells (LEFT, RIGHT, UP, DOWN should all work).

![async](https://user-images.githubusercontent.com/771170/198097300-b6e2e0dd-c789-44f0-ae37-f05d4ac1a373.gif)
